internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -363,6 +408,128 @@ func (cf CloudFormation) executeAndRenderChangeSet(in *executeAndRenderChangeSet
 	return nil
 }
 
+func (cf CloudFormation) waitForSignalAndHandleInterrupt(ctx context.Context, cancelFn context.CancelFunc, sigCh chan os.Signal, stackName string) error {
+	for {
+		select {
+		case <-sigCh:
+			cancelFn()
+			stopCatchSignals(sigCh)
+			stackDescr, err := cf.cfnClient.Describe(stackName)
+			if err != nil {
+				return fmt.Errorf("describe stack %s: %w", stackName, err)
+			}
+			switch aws.StringValue(stackDescr.StackStatus) {
+			case sdkcloudformation.StackStatusCreateInProgress:
+				log.Infof(`Received Interrupt for Ctrl-C.
+Pressing Ctrl-C again will exit immediately but the deletion of stack %s still happens
+`, stackName)
+				description := fmt.Sprintf("Delete stack %s", stackName)
+				if err := cf.deleteAndRenderStack(stackName, description, func() error {
+					return cf.cfnClient.DeleteAndWait(stackName)
+				}); err != nil {
+					return err
+				}
+				return &ErrStackDeletedOnInterrupt{stackName: stackName}
+			case sdkcloudformation.StackStatusUpdateInProgress:
+				log.Infof(`Received Interrupt for Ctrl-C.
+Pressing Ctrl-C again will exit immediately but stack %s rollback will continue
+`, stackName)
+				description := fmt.Sprintf("Canceling stack update %s", stackName)
+				if err := cf.cancelUpdateAndRender(&cancelUpdateAndRenderInput{
+					stackName:   stackName,
+					description: description,
+					cancelUpdateFn: func() error {
+						return cf.cfnClient.CancelUpdateStack(stackName)
+					},
+				}); err != nil {
+					return err
+				}
+				return &ErrStackUpdateCanceledOnInterrupt{stackName: stackName}
+			}
+			return nil
+		case <-ctx.Done():
+			stopCatchSignals(sigCh)
+			return nil
+		}
+	}
+}
+
+type cancelUpdateAndRenderInput struct {
+	stackName      string
+	description    string
+	cancelUpdateFn func() error
+}
+
+func (cf CloudFormation) cancelUpdateAndRender(in *cancelUpdateAndRenderInput) error {
+	stackDescr, err := cf.cfnClient.Describe(in.stackName)
+	if err != nil {
+		return fmt.Errorf("describe stack %s: %w", in.stackName, err)
+	}
+	if stackDescr.ChangeSetId == nil {
+		return fmt.Errorf("ChangeSetID not found for stack %s", in.stackName)
+
+	}
+	waitCtx, cancelWait := context.WithTimeout(context.Background(), waitForStackTimeout)
+	defer cancelWait()
+	g, ctx := errgroup.WithContext(waitCtx)
+	g.Go(in.cancelUpdateFn)
+	renderer, err := cf.createChangeSetRenderer(g, ctx, aws.StringValue(stackDescr.ChangeSetId), in.stackName, in.description, progress.RenderOptions{})
+	if err != nil {
+		return err
+	}
```

Comment:
could we move this below 480? just to keep the format similar
```
g.Go()
g.Go()

something else

g.Wait()
```

Reply:
`renderer` is getting passed as input parameter to one of the go routine though 🤔 
```_, err := progress.Render(ctx, progress.NewTabbedFileWriter(cf.console), renderer)```

So I will move this above both the go routines. I looks something like this.
```
renderer, err := cf.createChangeSetRenderer(g, ctx, aws.StringValue(stackDescr.ChangeSetId), in.stackName, in.description, progress.RenderOptions{})
g.Go()
g.Go(
_, err := progress.Render(ctx, progress.NewTabbedFileWriter(cf.console), renderer)
)